xray/commands/curation/audit.go

```diff
@@ -0,0 +1,508 @@
+package curation
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"github.com/jfrog/gofrog/parallel"
+	rtUtils "github.com/jfrog/jfrog-cli-core/v2/artifactory/utils"
+	"github.com/jfrog/jfrog-cli-core/v2/utils/coreutils"
+	audit "github.com/jfrog/jfrog-cli-core/v2/xray/commands/audit/generic"
+	cmdUtils "github.com/jfrog/jfrog-cli-core/v2/xray/commands/utils"
+	"github.com/jfrog/jfrog-cli-core/v2/xray/utils"
+	"github.com/jfrog/jfrog-client-go/artifactory"
+	"github.com/jfrog/jfrog-client-go/auth"
+	clientutils "github.com/jfrog/jfrog-client-go/utils"
+	"github.com/jfrog/jfrog-client-go/utils/errorutils"
+	"github.com/jfrog/jfrog-client-go/utils/io/httputils"
+	"github.com/jfrog/jfrog-client-go/utils/log"
+	xrayUtils "github.com/jfrog/jfrog-client-go/xray/services/utils"
+	"net/http"
+	"os"
+	"path/filepath"
+	"regexp"
+	"sort"
+	"strings"
+	"sync"
+)
+
+const (
+	// Blocked status presents the unapproved requests to curation service.
+	blocked                = "blocked"
+	BlockingReasonPolicy   = "Policy violations"
+	BlockingReasonNotFound = "Package pending update"
+
+	totalConcurrentRequests = 10
+	directRelation          = "direct"
+	indirectRelation        = "indirect"
+
+	BlockMessageKey  = "jfrog packages curation"
+	NotBeingFoundKey = "not being found"
+
+	extractPoliciesRegexTemplate = "({.*?})"
+)
+
+var extractPoliciesRegex *regexp.Regexp
```

Comment:
It is best to have this variable as a private property of the CurationAuditCommand struct, than to have it as just a variable.